python/cmsis_svd/parser.py

```diff
@@ -363,375 +353,486 @@ def for_mcu(cls, package_root, mcu):
+
+    def _parse_address_block(
+        self, address_block_node: Element
+    ) -> Union[SVDAddressBlock, None]:
+        usage = None
+        if usage_text := _get_text(address_block_node, 'usage'):
+            usage = self._parse_address_block_usage_type(usage_text)
+
+        if protection := _get_text(address_block_node, 'protection'):
+            protection = SVDProtectionType(protection)
+
+        return SVDAddressBlock(
+            offset=_get_int(address_block_node, 'offset'),
+            size=_get_int(address_block_node, 'size'),
+            usage=usage,
+            protection=protection
+        )
+
+    @staticmethod
+    def _parse_interrupt(interrupt_node: Element) -> SVDInterrupt:
+        return SVDInterrupt(
+            name=_get_text(interrupt_node, 'name'),
+            description=_get_text(interrupt_node, 'description'),
+            value=_get_int(interrupt_node, 'value'),
+        )
+
+    @staticmethod
+    def _parse_write_constraint(
+        write_constraint_node: Element
+    ) -> SVDWriteConstraint:
+        write_constraint_range = None
+        if (range_node := write_constraint_node.find('./range')) is not None:
+            write_constraint_range = SVDWriteConstraintRange(
+                minimum=_get_bool(range_node, 'minimum'),
+                maximum=_get_bool(range_node, 'maximum')
+            )
+
+        return SVDWriteConstraint(
+            write_as_read=_get_bool(write_constraint_node, 'writeAsRead'),
+            use_enumerated_values=_get_bool(write_constraint_node, 'useEnumeratedValues'),
+            range=write_constraint_range
+        )
+
+    @staticmethod
+    def _parse_usage_type(text_value: str) -> SVDEnumUsageType:
+        usage_text = text_value.strip().lower()
+        return SVDEnumUsageType(usage_text)
+
+    def _parse_enumerated_values(
+        self, enumerated_values_node: Element
+    ) -> SVDEnumeratedValues:
+        enum_values = [self._parse_enumerated_value(ev)
+                       for ev in enumerated_values_node.findall('./enumeratedValue')]
+
+        if usage := _get_text(enumerated_values_node, 'usage'):
+            usage = self._parse_usage_type(usage)
+
+        return SVDEnumeratedValues(
+            name=_get_text(enumerated_values_node, 'name'),
+            header_enum_name=_get_text(enumerated_values_node, 'headerEnumName'),
+            usage=usage,
+            enumerated_values=enum_values
+        )
+
+    def _parse_field(self, field_node: Element) -> Union[SVDField, SVDFieldArray]:
+        dim_element_group = self._parse_dim_element_group(field_node)
+
+        enum_values = None
+        if (enum_values_nodes := field_node.findall('./enumeratedValues')) is not None:
+            enum_values = [self._parse_enumerated_values(ev) for ev in enum_values_nodes]
+
+        mod_w_value = _get_text(field_node, 'modifiedWriteValues')
+        mod_w_value = (SVDModifiedWriteValuesType(mod_w_value) if mod_w_value
+                       else None)
```

Comment:
Given the line above is just as long too …
```suggestion
        mod_w_value = SVDModifiedWriteValuesType(mod_w_value) if mod_w_value else None
```

Reply:
Fixed.